File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/network/FintechService.java
package com.vladislavmyasnikov.courseproject.data.network;

import com.vladislavmyasnikov.courseproject.data.models.Login;
import com.vladislavmyasnikov.courseproject.data.models.Result;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface FintechService {

    @POST("signin")
    Call<Void> getAccess(@Body Login login);

    @GET("user")
    Call<Result> getUser(@Header("Cookie") String token);

    @GET("course/android_spring_2019/homeworks")
    Call<Result> getLectures(@Header("Cookie") String token);
}

File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/network/FintechService.kt
package com.vladislavmyasnikov.courseproject.data.network

import com.google.gson.annotations.SerializedName
import com.vladislavmyasnikov.courseproject.data.models.Lecture
import com.vladislavmyasnikov.courseproject.data.models.Profile
import com.vladislavmyasnikov.courseproject.data.models.Student
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST

interface FintechService {

    @POST("signin")
    fun getAccess(@Body login: Login): Call<Void>

    @GET("user")
    fun getProfile(@Header("Cookie") token: String): Call<ProfileInfo>

    @GET("course/android_spring_2019/homeworks")
    fun getLectures(@Header("Cookie") token: String): Call<Lectures>

    @GET("course/android_spring_2019/grades")
    fun getStudents(@Header("Cookie") token: String): Call<List<Students>>
}


class Login(@SerializedName("email") val email: String,
            @SerializedName("password") val password: String)

class CookieData(val token: String, val time: String)

class Lectures(@SerializedName("homeworks") val lectures: List<Lecture>)

class Students(@SerializedName("grades") val students: List<Student>)

class ProfileInfo(@SerializedName("user") val profile: Profile?)